packages/nimble-components/src/button/styles.ts

```diff
@@ -16,80 +16,83 @@ export const styles = buttonStyles.withBehaviors(
     appearanceBehavior(
         ButtonAppearance.outline,
         css`
-            :host(.primary) .control {
+            :host([appearance-variant='primary']) .control {
```

Comment:
They are not. I'm on the fence for the rationale but a bunch of things aren't like that classes actually exist on the templates, css properties actually exist, css values are the right format, etc.

My rationale is that CSS is already so weakly typed that the visual noise will end up hurting maintainability more than help prevent issues. If a property name changes / is renamed etc the matrix tests should result in a diff that we catch. I also thinking sweeping changes like being done in this PR are rare.

But I can usually be pretty easily convinced to try and make the typing better. But would need to understand the pros and cons better in this case. Maybe an even better long term fix is figuring out how to leverage SCSS with FAST and get more static validation that way.

I can't think of a particular action item at this time but let me know if there is something we should track.

Reply:
Yeah, it was more of an idle thought spurred by our usage of `tagFor` (which seems to be a similar sacrifice of readability in favor of maintainability) than a proposal that I want prioritize. I completely agree that it would make the CSS difficult to read. 

I don't think it's worth tracking so I'll close this thread.

Reply:
The tagFor in templates is actually a bit sneaky. My main reason for wanting it is so that it requires you to import that component if you leverage it's tag. Helps enforce the dependency tree by the source instead of by convention.

You don't have to make sure and do explicit tracking and typing `import '../icon/check';` when you actually depend on the class instance with `import {IconCheck} from '../icon/check';`.